core/src/main/java/org/kohsuke/stapler/Stapler.java

```diff
@@ -1152,6 +1152,20 @@ public FileItem convert(Class type, Object value) {
             }
         }, FileItem.class);
 
+        CONVERT_UTILS.register(new Converter() {
+            @Override
+            public org.apache.commons.fileupload.FileItem convert(Class type, Object value) {
+                if (value == null) {
+                    return null;
+                }
+                try {
+                    return org.apache.commons.fileupload.FileItem.fromFileUpload2FileItem(Stapler.getCurrentRequest().getFileItem(value.toString()));
+                } catch (ServletException | IOException e) {
+                    throw new ConversionException(e);
+                }
+            }
+        }, org.apache.commons.fileupload.FileItem.class);
```

Comment:
I honestly can't remember why I had to do this, but some test failed somewhere along the line back in August 2023 and I added this to fix it. If someone really wants to know the justification, I can try removing this and re-running hours worth of test suites to remember the original motivation, but I'm hoping nobody cares enough to ask this question.

Reply:
If I were to be interested in the answer, what test suites are you referring to? Core and/or BOM PCT?

Reply:
Yes, it was some PCT or ATH test if I remember correctly.